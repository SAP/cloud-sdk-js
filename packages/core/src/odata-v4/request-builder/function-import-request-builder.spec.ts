import nock from 'nock';
import {
  defaultDestination,
  defaultHost
} from '../../../test/test-util/request-mocker';
import {
  testFunctionImportMultipleParams,
  testFunctionImportSharedEntityReturnType
} from '../../../test/test-util/test-services/v4/test-service';

const serviceUrl = '/testination/sap/opu/odata/sap/API_TEST_SRV';

describe('FunctionImportRequestBuilder', () => {
  it('builds correct url for multiple parameters', async () => {
    const params = { stringParam: 'str1', nonNullableStringParam: 'str2' };

    const requestBuilder = testFunctionImportMultipleParams(params);

    const url = await requestBuilder.url(defaultDestination);
    const expected = expect.stringMatching(
      /TestFunctionImportMultipleParams\(.*StringParam.*\)?\$format=json/
    );
    expect(url).toEqual(expected);
    expect(url).toContain(`StringParam='${params.stringParam}'`);
    expect(url).toContain(
      `NonNullableStringParam='${params.nonNullableStringParam}'`
    );
  });

  it('throws an error when shared entity type is used as return type', async () => {
    nock(defaultHost)
      .get(`${serviceUrl}/TestFunctionImportSharedEntityReturnType()`)
      .query({ $format: 'json' })
      .reply(200, {});
    const requestBuilder = testFunctionImportSharedEntityReturnType({});

    await expect(
      requestBuilder.execute(defaultDestination)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
