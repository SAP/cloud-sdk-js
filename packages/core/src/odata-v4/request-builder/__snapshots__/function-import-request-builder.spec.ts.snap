// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionImportRequestBuilder throws an error when shared entity type is used as return type 1`] = `"Failed to build an entity from the response of the function import or action import: TestFunctionImportSharedEntityReturnType, because the entity type of the return type is shared by multiple entity sets."`;
